src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,185 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.InvalidCommandException;
+import duke.exceptions.InvalidTaskNumberException;
+import duke.exceptions.MissingDeadlineException;
+import duke.exceptions.MissingEventTimeException;
+import duke.exceptions.MissingTaskNameException;
+import duke.exceptions.MissingTaskNumberException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Makes sense of the user command
+ * Bridges communication between the user string and the tasks
+ */
+public class Parser {
+    /** TaskList object used by Duke **/
+    private static TaskList taskList;
+
+    /** Current user string **/
+    private static String userString;
+
+    /** Current user string split into words **/
+    private static String[] words;
+
+    /**
+     * Handles user input, and allowing interactions with tasks
+     *
+     * @param userString String that users enter
+     * @param taskList TaskList object used by Duke
+     * @throws DukeException If Duke does not recognize the format provided
+     */
+    public static String handle(String userString, TaskList taskList) throws DukeException {
+        Parser.userString = userString;
+        Parser.taskList = taskList;
+        String[] arr = userString.split(" ");
+        String firstWord = arr[0].toLowerCase();
+        Parser.words = arr;
+
+        switch (firstWord) {
+        case "hi":
+            return UI.hi();
+        case "hello":
+            return UI.hi();
+        case "bye":
+            taskList.saveTasksToStorage();
+            return UI.bye();
+        case "list":
+            return UI.list(taskList);
+        case "done":
+            return handleDone();
+        case "todo":
+            return handleTodo();
+        case "deadline":
+            return handleDeadline();
+        case "event":
+            return handleEvent();
+        case "delete":
+            return handleDelete();
+        case "find":
+            return handleFind();
+        default:
+            throw new InvalidCommandException("Invalid command");
```

Comment:
A neat way to handle all the different commands simultaneously. 👍

Reply:
Thanks! 👍